### ISSUE TITLE
FancyExtruderGenerator removes the partitioner

### ISSUE THREAD

**Issue by GiudGiud** [idaholab/moose#18696]
## Bug Description
The fancy extruder generator starts from a new mesh base and never copies over the input partitioner, nor changes the mesh base for the partitioner. 
So the partitioner is gone. And MOOSE is unhappy

Since neither operation (changing partitioner of a mesh, changing mesh base of a partitioner) are currently possible in libmesh, I think the partitioner needs to be created later with the final mesh generator mesh?
Rn the order of execution:
```
[DBG][ACT] TASK (         add_partitioner) TYPE (               PartitionerAction) NAME (     Partitioner) Memory usage 42MB
[DBG][ACT] TASK (      add_mesh_generator) TYPE (          AddMeshGeneratorAction) NAME (            core) Memory usage 42MB
[DBG][ACT] TASK (      add_mesh_generator) TYPE (          AddMeshGeneratorAction) NAME (         extrude) Memory usage 42MB
[DBG][ACT] TASK (      add_mesh_generator) TYPE (          AddMeshGeneratorAction) NAME (boundary_removal) Memory usage 42MB
[DBG][ACT] TASK (      add_mesh_generator) TYPE (          AddMeshGeneratorAction) NAME (     boundaries1) Memory usage 42MB
[DBG][ACT] TASK (      add_mesh_generator) TYPE (          AddMeshGeneratorAction) NAME (     boundaries2) Memory usage 42MB
[DBG][ACT] TASK (      add_mesh_generator) TYPE (          AddMeshGeneratorAction) NAME (     boundaries3) Memory usage 42MB
[DBG][ACT] TASK (      add_mesh_generator) TYPE (          AddMeshGeneratorAction) NAME (     boundaries4) Memory usage 42MB
[DBG][ACT] TASK (      add_mesh_generator) TYPE (          AddMeshGeneratorAction) NAME (     boundaries5) Memory usage 42MB
```

thoughts @friedmud ?

## Steps to Reproduce
Try to use both a Partitioner and a FancyExtruderGenerator

## Impact
Stopping me from running BEAVRS

**Comment by GiudGiud** [comment]
Fixed by #18696

**Comment by friedmud** [comment]
I think you mean #18735 👍
